symfony/plugins/orangehrmAdminPlugin/Api/LanguageAPI.php

```diff
@@ -0,0 +1,239 @@
+        return new EndpointGetAllResult(
+            LanguageModel::class,
+            $languages,
+            new ParameterBag([CommonParams::PARAMETER_TOTAL => $count])
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetAll(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            ...$this->getSortingAndPaginationParamsRules(LanguageSearchFilterParams::ALLOWED_SORT_FIELDS)
+        );
+    }
+
+    /**
+     * @inheritDoc
+     * @return EndpointCreateResult
+     * @throws Exception
+     */
+    public function create(): EndpointCreateResult
+    {
+        // TODO:: Check data group permission
+        $languages = $this->saveLanguage();
+
+        return new EndpointCreateResult(LanguageModel::class, $languages);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForCreate(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(self::PARAMETER_NAME),
+        );
+    }
+
+    /**
+     * @return Language
+     * @throws RecordNotFoundException
+     * @throws \OrangeHRM\Core\Exception\DaoException
+     * @throws Exception
+     */
+    public function saveLanguage(): Language
+    {
+        $id = $this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_ATTRIBUTE, CommonParams::PARAMETER_ID);
+        $name = $this->getRequestParams()->getString(RequestParams::PARAM_TYPE_BODY, self::PARAMETER_NAME);
+        if (!empty($id)) {
+            $language = $this->getLanguageService()->getLanguageById($id);
+            if ($language == null) {
+                throw new RecordNotFoundException();
+            }
+        } else {
+            $language = new Language();
+        }
+
+        $language->setName($name);
+        return $this->getLanguageService()->saveLanguage($language);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForSaveLanguage(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(CommonParams::PARAMETER_ID),
+            new ParamRule(self::PARAMETER_NAME),
+        );
+    }
+
+    /**
+     * @inheritDoc
+     * @return EndpointUpdateResult
+     * @throws Exception
+     */
+    public function update(): EndpointUpdateResult
+    {
+        // TODO:: Check data group permission
+        $languages = $this->saveLanguage();
+
+        return new EndpointUpdateResult(LanguageModel::class, $languages);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForUpdate(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
+                CommonParams::PARAMETER_ID,
+                new Rule(Rules::POSITIVE)
+            ),
+            new ParamRule(self::PARAMETER_NAME),
+        );
+    }
+
+    /**
+     *
+     * @return EndpointDeleteResult
+     * @throws Exception
+     */
```

Comment:
```php
    /**
     * @inheritDoc
     */
```

Reply:
@RajithaKumara I don't understand this ? should I remove it, it was auto generated. So i kept as it is